### PR TITLE
Adding_API_for_alcohol_prohibition

### PR DESCRIPTION
**Pull request by damn-dvlpr**
### Description:
   
    -Add alcohol_prohibition Property to data.json
    -Fix existing tests
    -Add getCountriesByAlcoholProhibition() Method
    -Add getCountriesByAlcoholProhibition() Tests
    -Document getCountriesByAlcoholProhibition() Method
    -Update ReadMe, changed the expected results of various APIs to incorporate new attribute.

### Types of changes

- [ ]  Bug fix (non-breaking change which fixes an issue)

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] Added the alcoholProhitibion API

- [x] I have updated the documentation accordingly.

- [x] I have added tests to cover my changes.

- [x] All new and existing tests passed.

- [x] updated Readme to incorporate new attribute

### CODE REVIEW

File: spec/getCountriesSpec.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ describe("The index", () => {
         phone_code: "+93",
         flag: "https://flagpedia.net/data/flags/h80/af.png",
         drive_direction: "right",
+        alcohol_prohibition: 'nationwide'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:28]:
```suggestion
        alcohol_prohibition: 'nationwide',
```

File: spec/getCountriesSpec.js [hunk 2]
```diff
@@ -40,6 +41,7 @@ describe("The index", () => {
         phone_code: "+355",
         flag: "https://flagpedia.net/data/flags/h80/al.png",
         drive_direction: "right",
+        alcohol_prohibition: 'none'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:44]:
```suggestion
        alcohol_prohibition: 'none',
```

File: spec/getCountriesSpec.js [hunk 3]
```diff
@@ -61,6 +63,7 @@ describe("The index", () => {
         phone_code: "+387",
         flag: "https://flagpedia.net/data/flags/h80/ba.png",
         drive_direction: "right",
+        alcohol_prohibition: 'none'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:66]:
```suggestion
        alcohol_prohibition: 'none',
```

File: spec/getCountriesSpec.js [hunk 4]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
       ).toBeTrue();
     });
   });
-});
+  
+  describe("The getCountriesByAlcoholProhibition where parameter is none", () => {
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:142]: I think that different argument for same _API_ Method is the same _test_ but different _test case_. We should have a single test (`describe()`) with multiple test cases (`it()`) for specific _API_ Method.

You can have a look at the other tests to _"match the pattern"_ 😉

File: spec/getCountriesSpec.js [hunk 5]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
-});
+  
+  describe("The getCountriesByAlcoholProhibition where parameter is none", () => {
+    it("returns the total number of countries where no alcohol prohibition is applicable", () => {
+      expect(countryApi.getCountriesByAlcoholProhibition("none").length).toEqual(176);
+    });
+  });
+
+  describe("The getCountriesByAlcoholProhibition where parameter is nationwide", () => {
+    const results=countryApi.getCountriesByAlcoholProhibition("nationwide");
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:149]:
```suggestion
    const results = countryApi.getCountriesByAlcoholProhibition("nationwide");

```

File: spec/getCountriesSpec.js [hunk 6]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+    const expectedResults=[
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:150]:
```suggestion
    const expectedResults = [

```

File: spec/getCountriesSpec.js [hunk 7]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+      {
+        country: 'afghanistan',
+        capital: 'kabul',
+        currency: 'afghani',
+        native_language: [ 'dari persian', 'pashto' ],
+        famous_for: 'rugs, taliban',
+        phone_code: '+93',
+        flag: 'https://flagpedia.net/data/flags/h80/af.png',
+        drive_direction: 'right',
+        alcohol_prohibition: 'nationwide'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:160]:
```suggestion
        alcohol_prohibition: 'nationwide',
```

File: spec/getCountriesSpec.js [hunk 8]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+  });
+
+  describe("The getCountriesByAlcoholProhibition where parameter is nationwide", () => {
+    const results=countryApi.getCountriesByAlcoholProhibition("nationwide");
+    const expectedResults=[
+      {
+        country: 'afghanistan',
+        capital: 'kabul',
+        currency: 'afghani',
+        native_language: [ 'dari persian', 'pashto' ],
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:155]:
```suggestion
        native_language: ['dari persian', 'pashto'],

```

File: spec/getCountriesSpec.js [hunk 9]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+        phone_code: '+93',
+        flag: 'https://flagpedia.net/data/flags/h80/af.png',
+        drive_direction: 'right',
+        alcohol_prohibition: 'nationwide'
+      },
+      {
+        country: 'brunei',
+        capital: 'bandar seri begawan',
+        currency: 'brunei dollar',
+        native_language: [ 'malay' ],
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:166]:
```suggestion
        native_language: ['malay'],

```

File: spec/getCountriesSpec.js [hunk 10]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+        famous_for: 'gorgeous mosques and islamic architecture',
+        phone_code: '+673',
+        flag: 'https://flagpedia.net/data/flags/h80/bn.png',
+        drive_direction: 'left',
+        alcohol_prohibition: 'nationwide'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:171]:
```suggestion
        alcohol_prohibition: 'nationwide',
```

File: spec/getCountriesSpec.js [hunk 11]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+      },
+      {
+        country: 'djibouti',
+        capital: 'djibouti',
+        currency: 'djiboutian franc',
+        native_language: [ 'arabic', 'french' ],
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:177]:
```suggestion
        native_language: ['arabic', 'french'],
```

File: spec/getCountriesSpec.js [hunk 12]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+        famous_for: 'amazing landscapes, limestone vents and flamingos',
+        phone_code: '+253',
+        flag: 'https://flagpedia.net/data/flags/h80/dj.png',
+        drive_direction: 'right',
+        alcohol_prohibition: 'nationwide'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:182]:
```suggestion
        alcohol_prohibition: 'nationwide',
```

File: spec/getCountriesSpec.js [hunk 13]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+      {
+        country: 'iran',
+        capital: 'tehran',
+        currency: 'rial',
+        native_language: [ 'persian' ],
+        famous_for: 'nuclear weapon, shia muslims, poetry, persian carpet',
+        phone_code: '+98',
+        flag: 'https://flagpedia.net/data/flags/h80/ir.png',
+        drive_direction: 'right',
+        alcohol_prohibition: 'nationwide'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:193]:
```suggestion
        alcohol_prohibition: 'nationwide',
```

File: spec/getCountriesSpec.js [hunk 14]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+      {
+        country: 'kuwait',
+        capital: 'kuwait city',
+        currency: 'kuwaiti dollar',
+        native_language: [ 'arabic', 'english' ],
+        famous_for: "victim of saddam's invasion",
+        phone_code: '965',
+        flag: 'https://flagpedia.net/data/flags/h80/kw.png',
+        drive_direction: 'right',
+        alcohol_prohibition: 'nationwide'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:204]:
```suggestion
        alcohol_prohibition: 'nationwide',
```

File: spec/getCountriesSpec.js [hunk 15]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+      },
+      {
+        country: 'libya',
+        capital: 'tripoli',
+        currency: 'libyan dinar',
+        native_language: [ 'arabic' ],
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:210]:
```suggestion
        native_language: ['arabic'],
```

File: spec/getCountriesSpec.js [hunk 16]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+        famous_for: 'sahara desert',
+        phone_code: '+218',
+        flag: 'https://flagpedia.net/data/flags/h80/ly.png',
+        drive_direction: 'right',
+        alcohol_prohibition: 'nationwide'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:215]:
```suggestion
        alcohol_prohibition: 'nationwide',
```

File: spec/getCountriesSpec.js [hunk 17]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+      {
+        country: 'mauritania',
+        capital: 'nouakchott',
+        currency: 'ouguiya',
+        native_language: [ 'arabic' ],
+        famous_for: 'large reserves of iron ore, copper, and gypsum',
+        phone_code: '+222',
+        flag: 'https://flagpedia.net/data/flags/h80/mr.png',
+        drive_direction: 'right',
+        alcohol_prohibition: 'nationwide'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:226]:
```suggestion
        alcohol_prohibition: 'nationwide',
```

File: spec/getCountriesSpec.js [hunk 18]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+      },
+      {
+        country: 'saudi arabia',
+        capital: 'riyadh',
+        currency: 'saudi riyal',
+        native_language: [ 'arabic' ],
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:232]:
```suggestion
        native_language: ['arabic'],
```

File: spec/getCountriesSpec.js [hunk 19]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+        famous_for: 'strict rules, wahhabism, mecca and madina',
+        phone_code: '+966',
+        flag: 'https://flagpedia.net/data/flags/h80/sa.png',
+        drive_direction: 'right',
+        alcohol_prohibition: 'nationwide'
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:237]:
```suggestion
        alcohol_prohibition: 'nationwide',
```

File: spec/getCountriesSpec.js [hunk 20]
```diff
@@ -135,4 +138,120 @@ describe("The index", () => {
...
+        phone_code: '+252',
+        flag: 'https://flagpedia.net/data/flags/h80/so.png',
+        drive_direction: 'right',
+        alcohol_prohibition: 'nationwide'
+      }
+    ];
+    it("returns the details of all countries where nationwide alcohol prohibition is applicable", () => {
+      expect(
+        testUtils.countryArraysAreEqual(results, expectedResults)
+      ).toBeTrue();
```
Comment (sthiepaan) [inline spec/getCountriesSpec.js:254]:
```suggestion
       expect(results).toEqual(expectedResults); 
```

File: package-lock.json [hunk 21]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "world-countries-capitals",
-	"version": "3.1.0",
+	"version": "3.2.1",
```
Comment (sthiepaan) [inline package-lock.json:3]: Please revert that change from this _Pull Request_. It should not be a part of this feature.
(I will have a look at it later)

**Review by sthiepaan — CHANGES_REQUESTED** [review]
As I understood you @damn-dvlpr, this was your first tests that you prepared. Nice job! 🎉

Please have a look and apply changes/suggestions. After you finish - instead of writing a comment - you can click on little circle icon in Reviewers section (located at top right of the Pull Request) near reviewer nickname 😉

File: spec/getCountriesSpec.js (hunk 4 above)
Comment (damn-dvlpr) [reply spec/getCountriesSpec.js:142]: Sure thing! will do that.

File: package-lock.json (hunk 21 above)
Comment (damn-dvlpr) [reply package-lock.json:3]: Sorry but I don't think i got what's to be done here.

**Comment by damn-dvlpr** [comment]
@sthiepaan I don't think the little button for re-review is working, cause I have clicked it many times and nothing seems to happen.

**Comment by damn-dvlpr** [comment]
@sthiepaan is the PR still missing something? if so, please let me know!

**Comment by damn-dvlpr** [comment]
@bhatvikrant will you please review this PR?

**Comment by sthiepaan** [comment]
@damn-dvlpr  _"Re-review"_ feature is working. I know you clicked it "few times" - I got the notification about it. We will check it whenever some of us got time, you don't need to spam...  Please, be patient ✌️

**Review by sthiepaan — APPROVED** [review]
LGTM

**Review by bhatvikrant — CHANGES_REQUESTED** [review]
Hey @damn-dvlpr, all looks good. Just resolve the conflicts and i think it's good to go then!

**Review by sthiepaan — CHANGES_REQUESTED** [review]
I would like to merge this _Pull Request_ as next but unfortunately we struggle with some merge conflicts. Conflicts are in:
- `README.md`
- `data/data.json`
- `index.js`
- `spec/getCountriesSpec.js`

Could you resolve them @damn-dvlpr? 😉

**Comment by damn-dvlpr** [comment]
Those were a lot to resolve! Finally done them.

**Review by sthiepaan — APPROVED** [review]
LGTM 🚀

**Pull request merged**